docs/upgrading_to_sql_db_4.0.0.md

```diff
@@ -0,0 +1,178 @@
+# Upgrading to SQL DB 4.0.0
+
+The v4.0.0 release of SQL DB is a backward incompatible release. The MySQL/PostgreSQL read replica state has been migrated to using `for_each` and the replica configuration is now a list of individual replica configuration rather then all replicas sharing the same settings.
+
+In addition the deprecated `failover_*` variables used to manage the legacy MySQL failover instance for HA have been completely removed in favour of the new HA solution that was introduced in v3.2.0
+
+## Migration Instructions
+#### MySQL Failover Instance
+
+This migration causes downtime, please read the CloudSQL docs first: [Updating an instance from legacy to current high availability](https://cloud.google.com/sql/docs/mysql/configure-ha#update-from-legacy)
+
+1. Delete the failover instance via this module
+2. Set `availability_type = "REGIONAL"` - This will cause downtime as the master instance is restarted
+
+
+#### Additional Users and Databases
+`additional_databases` and `additional_users` has moved to using `for_each` which requires a state migration. In addition `project` and `instance` object fields have been removed since they are inferred automatically.
+
+```diff
+   additional_databases = [
+     {
+-      project   = var.project_id
+-      instance  = null
+       name      = "db1"
+       charset   = "utf8mb4"
+       collation = "utf8mb4_general_ci"
+
+   additional_users = [
+     {
+-      project  = var.project_id
+-      instance = null
+       name     = "user1"
+       password = "abcdefg"
+       host     = "localhost"
+     }
+   ]
+
+```
+
+#### Read Replicas
+The new `read_replicas` variable is used to manage all replica configuration. In order to migrate existing replicas you will need to perform a state migration.
+
+- You must have `read_replica_size` objects inside `read_replicas`
```

Comment:
Shouldn't this be "You must **not** have read_replica_size...." ?

The `read_replicas` array itself would be the size.